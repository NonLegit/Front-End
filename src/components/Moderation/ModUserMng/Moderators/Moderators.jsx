/* eslint-disable import/no-cycle */
/* eslint-disable react/jsx-no-constructed-context-values */
import { Typography } from '@mui/material';
import ErrorOutlineOutlinedIcon from '@mui/icons-material/ErrorOutlineOutlined';
import * as React from 'react';
import { useParams } from 'react-router-dom';
import {
  QueueBox, QueueText, ControlBar,
} from '../styles';
import { ModMainPage } from '../../ModerationMainPage/styles';
import LeavePopUp from './LeavePopUp/LeavePopUp';
import { RedditButton } from '../../styles';
import Invitation from './InvitationPopUp/InvitationPopUp';
import ModeratorsList from './NonEmptyModerator/ModeratorsList/ModeratorsList';

export const LeaveContext = React.createContext();
export const InvitationContext = React.createContext();

/**
 * moderators whole page
 * @component
 * @return {React.Component} - mdoerators page component
 */

function Moderator() {
  const { subReddit } = useParams();

  const [openLeave, setOpenLeave] = React.useState(false);
  const handleClickOpenLeave = () => { setOpenLeave(true); };
  const handleClickCloseLeave = () => { setOpenLeave(false); };

  const [openInvitation, setOpenInvitation] = React.useState(false);
  const handleClickOpenInvitation = () => { setOpenInvitation(true); };
  const handleClickCloseInvitation = () => { setOpenInvitation(false); };

  return (
    <ModMainPage>
      <LeaveContext.Provider value={{
        openLeave, handleClickCloseLeave,
      }}
      >
        <LeavePopUp />
      </LeaveContext.Provider>
      <InvitationContext.Provider value={{
        openInvitation, handleClickCloseInvitation,
      }}
      >
        <Invitation />
      </InvitationContext.Provider>
      <ControlBar>
        <RedditButton
          fontSize="14px"
          padding="0px 24px 0px 24px"
          fontWeight="bold"
          variant="outlined"
          sx={{ marginRight: '12px' }}
          onClick={handleClickOpenLeave}
        >
          Leave as mod
        </RedditButton>
        <RedditButton
          fontSize="14px"
          padding="0px 24px 0px 24px"
          fontWeight="bold"
          variant="contained"
          onClick={handleClickOpenInvitation}
        >
          Invite user as mode
        </RedditButton>
      </ControlBar>
      <QueueBox>
        <QueueText>
          <Typography variant="h6">
            Moderators of r/
            {subReddit}
          </Typography>
          <ErrorOutlineOutlinedIcon color="primary" />
        </QueueText>
      </QueueBox>
      <ModeratorsList />
    </ModMainPage>
  );
}

export default Moderator;
