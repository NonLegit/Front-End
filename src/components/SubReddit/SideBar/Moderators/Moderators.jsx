import MailOutlineIcon from '@mui/icons-material/MailOutline'; import {
  AboutCountainer, AboutString, CustomLink, Button, Container, Name, ViewAll, ViewAllContainer,
} from './style';
/**
 * Moderators section in sidebar
 * @component
 * @property {object} moderatoesName -list of object of moderators

 * @return {React.Component} - Moderators section in sidebar
 */
function Moderators(props) {
  const { moderatoesName } = props;
  return (
    <>
      <AboutCountainer>
        <AboutString>
          Moderators
        </AboutString>
      </AboutCountainer>
      <CustomLink>
        <Button variant="outlined" padding="4px" fontSize={15} fontWeight="bold" sx={{ m: 1.8 }}>
          <MailOutlineIcon sx={{ margin: '2px' }} />
          {' '}
          Message the Mods
        </Button>
      </CustomLink>
      { moderatoesName?.map((entity, index) => (
        <Container key={`${index + 0}`}>
          <Name to={`/user/${entity}`}>
            u/
            {entity}
          </Name>
        </Container>
      ))}
      <ViewAllContainer>
        <ViewAll>VIEW ALL MODERATORS</ViewAll>
      </ViewAllContainer>
    </>
  );
}

export default Moderators;
