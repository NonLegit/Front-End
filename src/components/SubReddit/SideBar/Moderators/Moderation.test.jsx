import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Moderators from './Moderators';
// test snapshot
test('test snapshot', async () => {
  const props = {
    moderatoesName: ['aa', 'bb'],
  };
  const tree = renderer.create(
    <Router>
      <Moderators {...props} />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
