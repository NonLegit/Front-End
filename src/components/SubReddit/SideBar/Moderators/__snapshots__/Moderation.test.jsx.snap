// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
Array [
  <div
    className="css-1f2ln3t"
  >
    <h2
      className="css-dgde1h"
    >
      Moderators
    </h2>
  </div>,
  <a
    className="css-vgri8"
    href="/"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root css-azyh4b-MuiButtonBase-root"
      disabled={false}
      fontSize={15}
      fontWeight="bold"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      padding="4px"
      tabIndex={0}
      type="button"
      variant="outlined"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ubwraf-MuiSvgIcon-root"
        data-testid="MailOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
        />
      </svg>
       
      Message the Mods
    </button>
  </a>,
  <div
    className="css-1w5iul1"
  >
    <a
      className="css-bz7ml9"
      href="/user/aa"
      onClick={[Function]}
    >
      u/
      aa
    </a>
  </div>,
  <div
    className="css-1w5iul1"
  >
    <a
      className="css-bz7ml9"
      href="/user/bb"
      onClick={[Function]}
    >
      u/
      bb
    </a>
  </div>,
  <div
    className="css-e3tcub"
  >
    <a
      className="css-qq790j"
      href="/"
      onClick={[Function]}
    >
      VIEW ALL MODERATORS
    </a>
  </div>,
]
`;
